### DISCUSSION TITLE
How to add a new environment variable programatically

### DISCUSSION THREAD

**Discussion by MarcusAloshy** [eclipse-embed-cdt/eclipse-plugins#505] [Managed Build Plug-ins Q&A]
I'm trying to add a new environment variable (Eg: HOME) like CWD and PWD, in project  C/C++, Build/Environment variables every time creating a new C project using C Managed build template?

**Comment by ilg-ul** [comment]
Use the CDT APIs, for example implement a class derived from `IConfigurationEnvironmentVariableSupplier` which in turn should define classes derived from `IBuildEnvironmentVariable`.

**Reply by MarcusAloshy** [reply]
Is there any way without patching the plugin?

**Comment by ilg-ul** [comment]
English is not my first language, but adding a new environment variable programatically (as the subject implies) means you do this in program right?

**Reply by promodkumar-ashling** [reply]
Yes, this worked fine thanks
